### PR TITLE
Automatically expand and scroll the blueprint tree when focusing on an item

### PR DESCRIPTION
**Pull request by abey79**
### What

Double-clicking on an item, e.g. in a space views, focuses on that item. When that happens, the blueprint tree UI will uncollapse and scroll as needed to bring said item to view. When that item is an entity (e.g. double-clicked in the streams view), then all occurrences of that entity are uncollapsed, and the view scrolls to the first one.

https://github.com/rerun-io/rerun/assets/49431240/93b54fb6-ad52-4880-babd-d1001732c4db

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5482/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5482/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5482/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5482)
- [Docs preview](https://rerun.io/preview/ce5c582ffbe0844840846373a467a6907fb7b895/docs) 
- [Examples preview](https://rerun.io/preview/ce5c582ffbe0844840846373a467a6907fb7b895/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_viewport/src/viewport_blueprint.rs [hunk 1]
```diff
@@ -439,26 +440,50 @@ impl ViewportBlueprint {
...
+    }
+
     /// Walk the subtree defined by the provided container id and call `visitor` for each
     /// [`Contents`].
     ///
-    /// Note: `visitor` is first called for the container passed in argument.
+    /// Note:
+    /// - `visitor` is first called for the container passed in argument
+    /// - `visitor`'s second argument contains the hierarchy leading to the visited contents, up to
+    ///   (and including) the container passed in argument
```
Comment (emilk) [inline crates/re_viewport/src/viewport_blueprint.rs:458]:
```suggestion
    /// - `visitor`'s second argument contains the hierarchy leading to the visited contents, from
    ///   (and including) the container passed in argument
```

File: crates/re_viewport/src/viewport_blueprint_ui.rs [hunk 2]
```diff
@@ -81,6 +84,137 @@ impl Viewport<'_, '_> {
             });
     }
 
+    /// Expend all required items and compute which item we should scroll to.
+    fn handle_focused_item(&self, ctx: &ViewerContext<'_>, ui: &egui::Ui) -> Option<Item> {
+        ctx.focused_item.as_ref().and_then(|focused_item| {
```
Comment (emilk) [inline crates/re_viewport/src/viewport_blueprint_ui.rs:89]: prevent rightward drift with this weird trick

```suggestion
        let focused_item = ctx.focused_item.as_ref()?;
```

**Review by emilk — APPROVED** [review]
Nice!

File: crates/re_viewport/src/viewport_blueprint_ui.rs (hunk 2 above)
Comment (abey79) [reply crates/re_viewport/src/viewport_blueprint_ui.rs:89]: good one 👍🏻

**Pull request merged**